src/tests/tribol_comp_geom.cpp

```diff
@@ -551,61 +544,60 @@ TEST_F( CompGeomTest, codirectional_normals_3d )
    z[6] = -0.300001*element_thickness[1];
    z[7] = -0.300001*element_thickness[1];
 
-   // initialize tribol
-   tribol::CommType problem_comm = TRIBOL_COMM_WORLD;
-   tribol::initialize( dim, problem_comm );
-
    // register contact mesh
-   int meshId = 0;
-   tribol::IndexType conn[8] = {0,1,2,3,4,5,6,7}; // hard coded for a two face problem
-   tribol::registerMesh( meshId, numCells, numVerts, &conn[0], (int)(tribol::LINEAR_QUAD), &x[0], &y[0], &z[0] );
-
-   tribol::enableTimestepVote(true);
+   tribol::IndexT mesh_id = 0;
+   tribol::IndexT conn[8] = {0,1,2,3,4,5,6,7}; // hard coded for a two face problem
+   tribol::registerMesh( mesh_id, numCells, lengthNodalData, &conn[0], (int)(tribol::LINEAR_QUAD), &x[0], &y[0], &z[0], tribol::MemorySpace::Host );
 
-   real *fx;
-   real *fy;
-   real *fz; 
+   RealT *fx;
+   RealT *fy;
+   RealT *fz; 
    tribol::allocRealArray( &fx, lengthNodalData, 0. );
    tribol::allocRealArray( &fy, lengthNodalData, 0. );
    tribol::allocRealArray( &fz, lengthNodalData, 0. );
 
-   tribol::registerNodalResponse( meshId, fx, fy, fz );
+   tribol::registerNodalResponse( mesh_id, fx, fy, fz );
 
-   real *vx;
-   real *vy;
-   real *vz; 
-   real vel0 = -1.e-15;
+   RealT *vx;
+   RealT *vy;
+   RealT *vz; 
+   RealT vel0 = -1.e-15;
    tribol::allocRealArray( &vx, lengthNodalData, 0. );
    tribol::allocRealArray( &vy, lengthNodalData, 0. );
    tribol::allocRealArray( &vz, lengthNodalData, vel0 );
 
    // set second face to impacting velocity
-   real vel2 = 1.e-15;
+   RealT vel2 = 1.e-15;
    vz[4] = vel2;
    vz[5] = vel2;
    vz[6] = vel2;
    vz[7] = vel2;
 
-   tribol::registerNodalVelocities( meshId, vx, vy, vz );
+   tribol::registerNodalVelocities( mesh_id, vx, vy, vz );
 
-   real bulk_mod = 1.;
-   tribol::registerRealElementField( meshId, tribol::BULK_MODULUS, &bulk_mod );
-   tribol::registerRealElementField( meshId, tribol::ELEMENT_THICKNESS, &element_thickness[0] );
+   RealT bulk_mod[2];
+   bulk_mod[0] = 1.0;
+   bulk_mod[1] = 1.0;
+   tribol::registerRealElementField( mesh_id, tribol::BULK_MODULUS, bulk_mod );
```

Comment:
`bulk_mod` is already a pointer; it can be sent directly to `registerRealElementField()` as can `element_thickness`.  I changed `&element_thickness[0]` to `element_thickness` in the latest commit.